olive/evaluator/metric.py

```diff
@@ -91,6 +92,20 @@ def get_sub_type_info(self, info_name, no_priority_filter=True, callback=lambda
             sub_type_info[sub_type.name] = callback(getattr(sub_type, info_name))
         return sub_type_info
 
+    @validator("backend", always=True, pre=True)
+    def validate_backend(cls, v, values):
+        if values["type"] == MetricType.CUSTOM:
+            return None
+        from olive.evaluator.metric_backend import MetricBackend
+
+        assert v in MetricBackend.registry, f"Backend {v} is not in {list(MetricBackend.registry.keys())}"
+        if v == "huggingface_metrics":
+            try:
+                import evaluate  # noqa: F401
+            except ImportError:
+                raise ImportError("Please install the huggingface/evaluate package to use huggingface metrics.")
```

Comment:
it seems there are duplicated code in backend py file, could we reuse it?

Reply:
Updated.